### PR TITLE
Issue 6694 2.3.0

### LINKED ISSUE #6694: [Java][CXF] Enumerations defined for attributes of Data types in the YAML do not retain their values for the Java CXF Server stub
##### Description

When defining enumerations for attributes of Data types in the YAML you expect that the generated code respects their values. But, the CXF models are created in a way that when they are transformed into JSON or XML, the values are changed to all capital letters.

##### Swagger-codegen version

2.3.0

##### Swagger declaration file content or url

For example an attribute defined in the YAML using enums like this:

```yaml
TYPE:
        type: "string"
        description: "Option type"
        enum:
        - "Folder"
        - "Executable"
```
Generates the following Java server code to define the enumerations:

```java
public enum TYPEEnum {
    @XmlEnumValue("Folder")
    FOLDER(String.valueOf("Folder")),
    @XmlEnumValue("Executable")
    EXECUTABLE(String.valueOf("Executable"));
```
and the _get_ method is generated like this:

```java
  public String getTYPE() {
    return TYPE;
  }
```

##### Command line used for generation
..

##### Steps to reproduce
..

##### Related issues/PRs
..

##### Suggest a fix/enhancement

The CXF generation template for Data types could be modified to return the value of the enum instead of the enum itself.

By adding the following line to the pojo.mustache of JavaJaxRS/cxf

```
  {{#isEnum}}public String {{getter}}() {
    if({{name}} == null) {return null;}
    return {{name}}.value();
  }{{/isEnum}}{{^isEnum}}public {{{datatypeWithEnum}}} {{getter}}() {
    return {{name}};
  }{{/isEnum}}
```

The resulting Java code works without the above mentioned issue:
```java
  public String getTYPE() {
    if (TYPE == null) {
      return null;
    }
    return TYPE.value();
  }
```

### PR DESCRIPTION
**Pull request by gustavoapaz**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [x] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming langauge.

### Description of the PR

I respect the values of the enumerations when transforming JSON/XML to Java Objects in the CXF Server stub and Client generation. 

fix #6694

@bbdouglas, @JFCote, @sreeshas, @jfiala, @lukoyanov, @cbornet, @wing328

### CODE REVIEW

File: modules/swagger-codegen/src/main/resources/JavaJaxRS/cxf/pojo.mustache [hunk 1]
```diff
@@ -49,9 +49,18 @@ public class {{classname}} {{#parent}}extends {{{parent}}}{{/parent}} {
 {{#vendorExtensions.extraAnnotation}}
   {{{vendorExtensions.extraAnnotation}}}
 {{/vendorExtensions.extraAnnotation}}
-{{#useBeanValidation}}{{>beanValidation}}{{/useBeanValidation}}  public {{{datatypeWithEnum}}} {{getter}}() {
+{{#useBeanValidation}}{{>beanValidation}}{{/useBeanValidation}}  {{#isEnum}}{{^isListContainer}}{{^isMapContainer}}public {{datatype}} {{getter}}() {
+    if({{name}} == null) {
```
Comment (wing328) [inline modules/swagger-codegen/src/main/resources/JavaJaxRS/cxf/pojo.mustache]: I think there should be a space after `if` according to Java style guide.

**Comment by wing328** [comment]
@gustavoapaz Please rebase on the latest master to resolve the merge conflicts when you've time. Thanks.

**Comment by wing328** [comment]
@gustavoapaz looks like the rebase is not done correctly. I would suggest you to close this one and file a new one instead.

**Comment by gustavoapaz** [comment]
Moved to PR !6778

**Pull request closed without merging**